File: src/main/java/entities/gestionblog/Commentaire.java
package entities.gestionblog;

import java.sql.Date;

public class Commentaire {
    private int id_comment;
    private int user_id;
    private int id_post;
    private String content;
    private Date date;
    private int likes;

    public Commentaire(int id_comment, int user_id, int id_post, String content, Date date, int likes) {
        this.id_comment = id_comment;
        this.user_id = user_id;
        this.id_post = id_post;
        this.content = content;
        this.date = date;
        this.likes = likes;
    }

    public Commentaire() {
    }

    public int getId_comment() {
        return id_comment;
    }

    public void setId_comment(int id_comment) {
        this.id_comment = id_comment;
    }

    public int getUser_id() {
        return user_id;
    }

    public void setUser_id(int user_id) {
        this.user_id = user_id;
    }

    public int getId_post() {
        return id_post;
    }

    public void setId_post(int id_post) {
        this.id_post = id_post;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public int getLikes() {
        return likes;
    }

    public void setLikes(int likes) {
        this.likes = likes;
    }

    @Override
    public String toString() {
        return "Commentaire{" +
                "id_comment=" + id_comment +
                ", user_id=" + user_id +
                ", id_post=" + id_post +
                ", content='" + content + '\'' +
                ", date=" + date +
                ", likes=" + likes +
                '}';
    }
}

File: webapp/src/GymPlus/src/Entity/Commentaire.php
<?php

namespace App\Entity;

use App\Repository\CommentaireRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: CommentaireRepository::class)]
class Commentaire
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column(name: "id_comment")]
    private ?int $id = null;

    #[ORM\Column(name: "user_id", nullable: true)]
    private ?int $userId = null;

    #[ORM\Column(name: "id_post", nullable: true)]
    private ?int $postId = null;

    #[ORM\Column(name: "content", length: 255, nullable: true)]
    private ?string $content = null;

    #[ORM\Column(name: "date", type: Types::DATE_MUTABLE, nullable: true)]
    private ?\DateTimeInterface $date = null;

    #[ORM\Column(name: "likes", nullable: true)]
    private ?int $likes = null;

    // #[ORM\ManyToOne(inversedBy: 'comnts')]
    // private ?Post $post = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getUserId(): ?int
    {
        return $this->userId;
    }

    public function setUserId(?int $userId): static
    {
        $this->userId = $userId;

        return $this;
    }

    public function getPostId(): ?int
    {
        return $this->postId;
    }

    public function setPostId(?int $postId): static
    {
        $this->postId = $postId;

        return $this;
    }

    public function getContent(): ?string
    {
        return $this->content;
    }

    public function setContent(?string $content): static
    {
        $this->content = $content;

        return $this;
    }

    public function getDate(): ?\DateTimeInterface
    {
        return $this->date;
    }

    public function setDate(?\DateTimeInterface $date): static
    {
        $this->date = $date;

        return $this;
    }

    public function getLikes(): ?int
    {
        return $this->likes;
    }

    public function setLikes(?int $likes): static
    {
        $this->likes = $likes;

        return $this;
    }

    // public function getPost(): ?Post
    // {
    //     return $this->post;
    // }

    // public function setPost(?Post $post): static
    // {
    //     $this->post = $post;

    //     return $this;
    // }
}
